modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -186,18 +186,18 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
 
     hints
       .collectFirst { case h: Hint.Documentation => h }
-      .map { doc =>
-        val shapeLinesOpt = split(doc.docString).map(_.toList)
-        val memberLinesOpt = doc.memberDocs.flatMap {
+      .flatMap { doc =>
+        val shapeLinesOpt = doc.docString.map(split(_)).map(_.toList)
+        val memberLinesOpt = doc.memberDocs.map(m => m.flatMap {
           case (memberName, memberDoc) =>
             split(memberDoc) match {
               case Some(NonEmptyList(head, rest)) =>
                 s"@param $memberName $head" :: rest
               case None => Nil
             }
```

Comment:
help, I'm losing the game of aligning types! :P

No but seriously, I'm having trouble getting the docs in a `List` as `List(shapeLinesOpt, memberLinesOpt).flatten.flatten`, as in I don't know how to get unnest these options(shapeDocs and memberDocs)  using `map` or `flatMap` correctly.

Reply:
pro-tip: you shouldn't have to : the `lines()` dsl method is there to flatten stuff for you, no matter the level of nesting. The code should roughly look like  : 

```scala
hints.collectFirst { .... }.foldMap { doc => 
   lines(
      doc.docString.flatMap(split), 
      doc.memberDocs.combineAll.foldMap {
           case (memberName, memberDoc) => //... 
      } 
   ) 
}
```